packages/cli/src/commands/git/connect.ts

```diff
@@ -101,10 +121,194 @@ export default async function connect(
+    parsedOrg,
+    repo
+  );
+  if (typeof connect === 'number') {
+    return connect;
+  }
+  output.log(
+    `Connected ${formatProvider(provider)} repository ${chalk.cyan(repoPath)}!`
+  );
+  return 0;
+}
+
+async function connectArgWithLocalGit(
+  client: Client,
+  output: Output,
+  org: Org,
+  project: Project,
+  confirm: boolean,
+  gitConfig: Dictionary<any>,
+  repoArg: string
+) {
+  const parsedUrlArg = parseRepoUrl(repoArg);
+  if (!parsedUrlArg) {
+    output.error(
+      `Cannot parse URL ${repoArg}. Make sure you didn't make any typos.`
+    );
+    return 1;
+  }
+
+  const remoteUrls = pluckRemoteUrls(gitConfig);
+  if (remoteUrls) {
+    const shouldConnect = await promptConnectArg(
+      client,
+      output,
+      confirm,
+      repoArg,
+      remoteUrls
+    );
+    if (typeof shouldConnect === 'number') {
+      return shouldConnect;
+    }
+    if (shouldConnect) {
+      const { provider, org: parsedOrg, repo } = parsedUrlArg;
+      const repoPath = `${parsedOrg}/${repo}`;
+      output.log(`Connecting Git remote: ${link(repoArg)}`);
+      const connect = await checkExistsAndConnect(
+        client,
+        output,
+        confirm,
+        org,
+        project,
+        project.link,
+        provider,
+        repoPath,
+        parsedOrg,
+        repo
+      );
+      if (typeof connect === 'number') {
+        return connect;
+      }
+      output.log(
+        `Connected ${formatProvider(provider)} repository ${chalk.cyan(
+          repoPath
+        )}!`
+      );
+    }
+    return 0;
+  }
+  return await connectArg(client, output, confirm, org, project, repoArg);
+}
+
+async function promptConnectArg(
+  client: Client,
+  output: Output,
+  yes: boolean,
+  arg: string,
+  remoteUrls: Dictionary<string>
+) {
+  const multiple = Object.keys(remoteUrls).length > 1;
+  if (multiple) {
+    output.log(`Found multiple Git repositories in your local Git config:`);
+    printRemoteUrls(output, remoteUrls);
+  } else {
+    const url = Object.values(remoteUrls)[0];
+    const parsedUrl = parseRepoUrl(url);
+    if (!parsedUrl) {
+      output.error(
+        `Cannot parse URL ${url}. Make sure you didn't make any typos.`
+      );
+      return 1;
+    }
+    const parsedUrlArg = parseRepoUrl(arg);
+    if (!parsedUrlArg) {
+      output.error(
+        `Cannot parse URL ${arg}. Make sure you didn't make any typos.`
+      );
+      return 1;
+    }
+    if (JSON.stringify(parsedUrl) === JSON.stringify(parsedUrlArg)) {
+      return true;
+    }
+
+    output.log(
+      `Found a repository in your local Git Config: ${chalk.cyan(
+        Object.values(remoteUrls)[0]
+      )}`
+    );
+  }
+
+  let shouldConnect = yes;
+  if (!shouldConnect) {
+    shouldConnect = await confirm(
+      client,
+      `Do you still want to connect ${chalk.cyan(arg)}?`,
+      false
+    );
+    if (!shouldConnect) {
+      output.log(`Aborted. Repo not connected.`);
+    }
+  }
+  return shouldConnect;
+}
```

Comment:
**suggestion (non-blocking):** use clearer function and variable identifiers

I think the names of functions and variables around this area can be confusing. This function uses `arg` as a variable name, which I believe is the command-line provided repo url, which it needs to compare to `parsedUrl`, which I believe is the git-config provided repo url.

For example: If these were called `currentRepoUrl` (for the git config one) and `newRepoUrl`, I'd be able to tell them apart better.

I think we could make other names around this area better in the same way. I'd be happy to pair on it. It's certainly not blocking.

Reply:
Thanks for helping me get started! I think it's a bit better now.,